modules/ROOT/pages/dataweave-selectors.adoc

```diff
@@ -48,6 +47,402 @@ xref:dataweave-cookbook-extract-data#selector_metadata[Extract Data].
 
 |===
 
+== Rules for matching 
+
+All selectors that uses a `name` to select are composed of a `Namespace` and a `Local Part`. When using namespace it is only going to match a `Key` with that `Namespace` and that `Local Part`. If the `name` only has a `Local Part` it is going to match all the `Key` whoes `Local Part` are equal dispaite it has `Namespace` or not.
+
+
+
+== Dynamic Selector
+
+The way to do dynamic selection is by using the `[``]` way of selecting. 
+
+- Single Value: `payload[(nameExpression)]`
+- Multi Value: `payload[*(nameExpression)]`
+- Attribute: `payload[@(nameExpression)]`
+- Key Value: `payload[&(nameExpression)]`
+
+The `nameExpression` should return a `String` type expression.
+
+
+== Single Value Selector
+
+Selects the value of the first key value pair whoes key matches the given selector name. It can be applied to an Object or to an Array. When it is applied to an Array it will execute the selector to all the Object values that are inside and ignoring the ones that are not Object. If no key was matched it return `null`
+
+=== Example
+
+This example shows how selector works in a simple json object
+
+==== Input
+
+[source,json, linenums]
+----
+{
+    "name": "Data Weave"
+}
+----
+
+==== Source
+
+[source,DataWeave, linenums]
+----
+%dw 2.0
+output application/json
+---
+payload.name
+----
+
+==== Output
+
+[source,json, linenums]
+----
+"Data Weave"
+----
+
+=== Example
+
+When selector is applied to an Array it will apply the selection to every element.
+
+==== Input
+
+[source,json, linenums]
+----
+[{
+    "name": "Arg"
+},
+{
+    "name": "Japan"
+}
+]
+----
+
+==== Source
+
+[source,DataWeave, linenums]
+----
+%dw 2.0
+output application/json
+---
+payload.name
+----
+
+==== Output
+
+[source,json, linenums]
+----
+["Arg", "Japan"]
+----
+
+
+=== Example
+
+Shows how to select an xml with namespace or directly with local name
+
+==== Input
+
+[source,xml, linenums]
+----
+<root>
+
+<h:table xmlns:h="http://www.w3.org/TR/html4/">
+  <h:tr>
+    <h:td>Apples</h:td>
+    <h:td>Bananas</h:td>
+  </h:tr>
+</h:table>
+
+<f:table xmlns:f="https://www.w3schools.com/furniture">
+  <f:name>African Coffee Table</f:name>
+  <f:width>80</f:width>
+  <f:length>120</f:length>
+</f:table>
+
+</root>
+----
+
+==== Source
+
+[source,DataWeave, linenums]
+----
+%dw 2.0
+output application/xml
+ns html http://www.w3.org/TR/html4/
+ns furniture https://www.w3schools.com/furniture
+---
+root: {
+     a: payload.root.html#table,
+     b: payload.root.furniture#table
+ }    
+----
+
+==== Output
+
+[source,xml, linenums]
+----
+<?xml version='1.0' encoding='UTF-8'?>
+<root>
+  <a>
+    <h:tr xmlns:h="http://www.w3.org/TR/html4/">
+      <h:td>Apples</h:td>
+      <h:td>Bananas</h:td>
+    </h:tr>
+  </a>
+  <b>
+    <f:name xmlns:f="https://www.w3schools.com/furniture">African Coffee Table</f:name>
```

Comment:
It is weird that the output adds the namespace attribute to all f: elements but not to all he: elements. Is there a reason for this behavior.